core/scheduler/src/main/scala/org/apache/openwhisk/core/scheduler/queue/SchedulingDecisionMaker.scala

```diff
@@ -0,0 +1,244 @@
+package org.apache.openwhisk.core.scheduler.queue
+
+import akka.actor.{Actor, ActorSystem, Props}
+import org.apache.openwhisk.common.Logging
+import org.apache.openwhisk.core.entity.FullyQualifiedEntityName
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+class SchedulingDecisionMaker(
```

Comment:
This component decides whether to add more containers or not for the given action.
It uses the average execution time of an action.

There are three cases to calculate the average execution.
1. Initial execution.
In this case, we just assume the execution time as tick time(100ms by default).

2. Queue is newly created.
In this case, activations exist in the DB, but no data in the AverageRingBuffer.
So we fetch the average execution time from DB.

3. An action is continuously executed.
Each container sends the last execution time along with `FetchRequest` when it pulls a new activation from a queue.
And our scheduler keeps N execution times in the AverageRingBuffer. 
So we can easily calculate the average execution time of recent N executions.